saleor/graphql/discount/utils.py

```diff
@@ -54,6 +57,32 @@ def get_products_for_promotion(promotion: Promotion) -> ProductsQueryset:
     return Product.objects.filter(Exists(variants.filter(product_id=OuterRef("id"))))
 
 
+def get_products_for_recalculate_discounted_price(
+    products: ProductsQueryset,
+) -> ProductsQueryset:
+    """Get products that are marked for price recalculation."""
+    return products.filter(recalculate_discounted_price=True)
+
+
+def mark_products_of_promotion_for_recalculate_discounted_price(promotion_pk: UUID):
+    """Mark products of specific promotion for recalculate discounted prices."""
+    try:
+        promotion = Promotion.objects.get(pk=promotion_pk)
+    except ObjectDoesNotExist:
+        logging.warning(f"Cannot find promotion with id: {promotion_pk}.")
+        return
+    products = get_products_for_promotion(promotion)
+    if products:
+        products.update(recalculate_discounted_price=True)
+
+
+def mark_products_for_recalculate_discounted_price(product_ids: Iterable[int]):
+    """Mark products for recalculate discounted prices."""
+    products = Product.objects.filter(pk__in=product_ids)
+    if products:
```

Comment:
Do you need this `if` here? 
You could make 
```python
Product.objects.filter(pk__in=product_ids).update(recalculate_discounted_price=True)
```
Additionally you could skip the update action when `product_ids` is an empty list